src/objects/api/validators.py

```diff
@@ -88,3 +89,35 @@ def validate_data_attrs(value: str):
                 "Operator `%(operator)s` supports only dates and/or numeric values"
             ) % {"operator": operator}
             raise serializers.ValidationError(message, code=code)
+
+
+class GeometryValidator:
+    code = "geometry-not-allowed"
+    message = _("This object type doesn't support geometry")
+
+    def set_context(self, serializer):
+        """
+        This hook is called by the serializer instance,
+        prior to the validation call being made.
+        """
+        self.instance = getattr(serializer, "instance", None)
+
+    def __call__(self, attrs):
+        object_type = attrs.get("object_type") or self.instance.object_type
+        geometry = attrs.get("record", {}).get("geometry")
+
+        if not geometry:
+            return
+
+        client = object_type.service.build_client()
+        try:
+            response = client.retrieve("objecttype", url=object_type.url)
+        except ClientError as exc:
+
+            msg = f"Object type can not be retrieved: {exc.args[0]}"
+            raise ValidationError(msg)
+
+        has_geometry = response.get("hasGeometry")
```

Comment:
What happens if the objecttype API is an older version and this property is not there yet? The default should be to allow it because it was always allowed before.

Please add a test as well.

Reply:
Good catch! I made the default value `True` and added a test